### PR TITLE
Added ReverseRangeWithOptions and MultiReverseRangeWithOptions

### LINKED ISSUE #65: Support for TS.MREVRANGE
Further reference: https://oss.redislabs.com/redistimeseries/commands/#tsmrangetsmrevrange

### LINKED ISSUE #66: Support for TS.REVRANGE
Further reference: https://oss.redislabs.com/redistimeseries/commands/#tsrangetsrevrange

### PR DESCRIPTION
**Pull request by filipecosta90**
Based on user requests ( cc @stevesim101 ) I've quickly added ReverseRangeWithOptions and MultiReverseRangeWithOptions, and included examples on its usage:
- fixes #66 
- fixes #65 
This PR also extendes client tests on Range, RevRange, MultiRevRange

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -214,6 +214,25 @@ func (client *Client) RangeWithOptions(key string, fromTimestamp int64, toTimest
 	return
 }
 
+// ReverseRangeWithOptions - Query a timestamp range on a specific time-series in reverse order
+// args:
+// key - time-series key name
+// fromTimestamp - start of range. You can use TimeRangeMinimum to express the minimum possible timestamp.
+// toTimestamp - end of range. You can use TimeRangeFull or TimeRangeMaximum to express the maximum possible timestamp.
+// rangeOptions - RangeOptions options. You can use the default DefaultRangeOptions
+func (client *Client) ReverseRangeWithOptions(key string, fromTimestamp int64, toTimestamp int64, rangeOptions RangeOptions) (dataPoints []DataPoint, err error) {
```
Comment (gkorland) [inline client.go:223]: @filipecosta90 don't you feel like ReverseRangeWithOptions and RangeWithOptions have duplicate code? can't you create a generic method that gets "TS.REVRANGE"/"TS.RANGE" and make these two call it?

File: client.go [hunk 2]
```diff
@@ -248,6 +267,25 @@ func (client *Client) MultiRangeWithOptions(fromTimestamp int64, toTimestamp int
...
 
+// MultiReverseRangeWithOptions - Query a timestamp range across multiple time-series by filters, in reverse direction.
+// args:
+// fromTimestamp - start of range. You can use TimeRangeMinimum to express the minimum possible timestamp.
+// toTimestamp - end of range. You can use TimeRangeFull or TimeRangeMaximum to express the maximum possible timestamp.
+// mrangeOptions - MultiRangeOptions options. You can use the default DefaultMultiRangeOptions
+// filters - list of filters e.g. "a=bb", "b!=aa"
+func (client *Client) MultiReverseRangeWithOptions(fromTimestamp int64, toTimestamp int64, mrangeOptions MultiRangeOptions, filters ...string) (ranges []Range, err error) {
+	conn := client.Pool.Get()
+	defer conn.Close()
```
Comment (gkorland) [inline client.go:278]: @filipecosta90 same here, don't you feel you can avoid this code duplication?

File: client.go (hunk 1 above)
Comment (filipecosta90) [reply client.go:223]: @gkorland I my opinion, given that they are two distinct commands I would not merge the method on the client. I would avoid logic on the client code that either issues command A or B depending on a flag. WDYT?

File: client.go (hunk 2 above)
Comment (filipecosta90) [reply client.go:278]: I would strict the duplicate of code ( that is little ) due to the reasons above...

File: client.go (hunk 1 above)
Comment (filipecosta90) [reply client.go:214]: @gkorland now that I've checked your PR I fully get the point on it ( I was thinking about exposing the same method that would issue different commands ). Totally agree on internal code reuse :)

File: client.go (hunk 2 above)
Comment (filipecosta90) [reply client.go:275]: fixed in last commit :+1:

**Pull request merged**